### PR TITLE
Task-3

### PR DESCRIPTION
**Pull request by Oksana-Lopatkina**
Product-service is done.
Functions:
- getProductById
- getProductsList
- getProductsAvailableList
The template is aws-nodejs-typescript
Additional scope:
- async, ES6 modules
- separated functions
- auto created api documentation via plugin serverless-auto-swagger
- handled 404 error

### CODE REVIEW

**Comment by Oksana-Lopatkina** [comment]
Swagger - https://qrc6eg3r7d.execute-api.us-east-1.amazonaws.com/dev/dev/swagger

File: product-service/src/functions/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default as getProductsList } from './getProductsList';
```
Comment (NotanKh) [inline product-service/src/functions/index.ts:1]: looks useless

File: product-service/src/functions/getProductsList/handler.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    return formatJSONResponse({
+      data,
+      event,
+    });
+  } catch (error) {
+    return {
+      statusCode: 500,
+      headers: { "content-type": "application/json" },
```
Comment (NotanKh) [inline product-service/src/functions/getProductsList/handler.ts:28]: no cors headers here, will this work as expected?

File: product-service/src/functions/getProductsList/handler.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda";
+// import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { formatJSONResponse } from '@libs/api-gateway';
+import { middyfy } from '@libs/lambda';
+
+// import schema from './schema';
+
+import { Product } from '../../model/product';
+import productsData from '../../store/products.json';
+const getProductsList = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
```
Comment (NotanKh) [inline product-service/src/functions/getProductsList/handler.ts:10]: is this the same that `getProductsAvailableList`?

File: product-service/src/functions/index.ts (hunk 1 above)
Comment (Oksana-Lopatkina) [reply product-service/src/functions/index.ts:1]: I fixed this in the next PR

File: product-service/src/functions/getProductsList/handler.ts (hunk 2 above)
Comment (Oksana-Lopatkina) [reply product-service/src/functions/getProductsList/handler.ts:28]: This is also fixed in the next PR

File: product-service/src/functions/getProductsList/handler.ts (hunk 3 above)
Comment (Oksana-Lopatkina) [reply product-service/src/functions/getProductsList/handler.ts:10]: No, the function getProductsAvailableList returns a filtered array with products whose stock count is greater than 0

**Review by NotanKh — APPROVED** [review]

**Pull request merged**